### PR TITLE
[math] Fix Math.Pow (1, Double.MaxValue) != 1 on Linux 32bits

### PR DESCRIPTION
**Pull request by ludovic-henry**

### CODE REVIEW

**Comment by alexrp** [comment]
The commit message needs some more details as to why this is needed.

**Comment by esdrubal** [comment]
build

**Comment by akoeplinger** [comment]
context: https://github.com/mono/mono/commit/92a0ac52a7018199f39f01625eb1025baf564743#commitcomment-12706481 (but I agree the explanation should be in the commit message).

Thanks for fixing it Ludovic, let's see if we finally get a green i386 build on jenkins :)

File: mono/metadata/sysmath.c [hunk 1]
```diff
@@ -243,7 +252,13 @@ ves_icall_System_Math_Pow (gdouble x, gdouble y)
 		 * a return value of 1 for that case.  We need to return
 		 * the same result as Windows. */
 		result = NaN.d;
-	} else {
+	}
+#if defined(__linux__) && SIZEOF_VOID_P == 4
+	else if (isplusone (x) && isplusmaxvalue (y)) {
+		result = POne.d;
```
Comment (akoeplinger) [inline mono/metadata/sysmath.c]: Why don't we just always short-circuit when x is 1, since 1^n = 1 for all n?

**Comment by esdrubal** [comment]
:thumbsup:

**Comment by ludovic-henry** [comment]
Thank you! :)

**Comment by akoeplinger** [comment]
@ludovic-henry looks like now that this one is fixed, another test break is uncovered on [i386](https://jenkins.mono-project.com/job/test-mono-mainline/2382/label=debian-i386/testReport/junit/MonoTests.System/MathTest/TestPow/).

```
MESSAGE:
#47
  Expected: 0.0d
  But was:  NaN

+++++++++++++++++++
STACK TRACE:
  at MonoTests.System.MathTest.TestPow () [0x008e1] in /var/lib/jenkins/workspace/test-mono-mainline/label/debian-i386/mcs/class/corlib/Test/System/MathTest.cs:585
  at (wrapper managed-to-native) System.Reflection.MonoMethod:InternalInvoke (System.Reflection.MonoMethod,object,object[],System.Exception&)
  at System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00038] in /var/lib/jenkins/workspace/test-mono-mainline/label/debian-i386/mcs/class/corlib/System.Reflection/MonoMethod.cs:295
```

**Pull request merged**